### PR TITLE
Clean-up from fixing test infrastructure

### PR DESCRIPTION
**Pull request by mschreter**
This PR solves the issues suggested in the review of PR #115.

### CODE REVIEW

File: include/meltpooldg/interface/scratch_data.hpp [hunk 1]
```diff
@@ -46,15 +46,17 @@ namespace MeltPoolDG
     using BlockVectorType = LinearAlgebra::distributed::BlockVector<number>;
 
   public:
-    ScratchData(const unsigned int max_verbosity_level = 0, const bool do_matrix_free = true)
+    ScratchData(const unsigned int max_verbosity_level = 0,
+                const bool         do_matrix_free      = true,
+                const MPI_Comm     mpi_communicator    = MPI_COMM_WORLD)
```
Comment (peterrum) [inline include/meltpooldg/interface/scratch_data.hpp:51]: I would prefer if we would be explicit regarding the communicator (this is normally our policy in deal.II). I think we have access to the communicator where we create `ScratchData`, don't we?

**Review by peterrum — APPROVED** [review]
Thanks a lot! My comments did not imply that you should do this ;)

**Comment by mschreter** [comment]
> Thanks a lot! My comments did not imply that you should do this ;)

I am so happy right now, so I decided to fix this immediately :)

File: include/meltpooldg/interface/scratch_data.hpp (hunk 1 above)
Comment (mschreter) [reply include/meltpooldg/interface/scratch_data.hpp:51]: Okay, maybe we can discuss this another time. I''ve opened an issue relating to this topic https://github.com/MeltPoolDG/MeltPoolDG/issues/119.

**Pull request merged**